src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -133,20 +133,22 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
                     }
 
                     if (this.instrumentIdentityToMetric.TryGetValue(instrumentIdentity, out var existingMetric))
-                    {
-                        metrics.Add(existingMetric);
-                        continue;
-                    }
-
-                    if (this.metricStreamNames.Contains(metricStreamName))
                     {
                         OpenTelemetrySdkEventSource.Log.DuplicateMetricInstrument(
```

Comment:
This log must only be inside the if (metrics.Count==0) part.

Reply:
On second thought, I removed this log message. I think it is unnecessary. It is not an issue of a duplicate conflicting metric instrument. The `if (metrics.Count == 0)` case is meant to handle instruments mapping to identical metric streams (i.e., same name, kind, description, etc) - so it is safe to aggregate together.

For example, this test covers this case

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3bf47f835f2899e13df88f01a4870228db93df41/test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs#L503-L524

Both `instrument1` and `instrument2` match the first view, the metric stream is identical, so they can be aggregated together. The second view is ignored, and a log message is produced in this case.